src/types/plugin-inputs.ts

```diff
@@ -0,0 +1,58 @@
+import { EmitterWebhookEvent as WebhookEvent, EmitterWebhookEventName as WebhookEventName } from "@octokit/webhooks";
+import { SupportedEventsU } from "./context";
+import { StaticDecode, Type as T } from "@sinclair/typebox";
+import { DiscriminatedUnionValidator } from "typebox-validators/discriminated";
+import { StandardValidator } from "typebox-validators";
+
+export interface PluginInputs<T extends WebhookEventName = SupportedEventsU> {
+  stateId: string;
+  eventName: T;
+  eventPayload: WebhookEvent<T>["payload"];
+  settings: PermitGenerationSettings;
+  authToken: string;
+  ref: string;
+}
+
+const permitRequestSchema = T.Union([
+  T.Object({
+    kind: T.Literal("ERC20"), // Discriminant key
+    type: T.Literal("ERC20"), // should remove this ideally
```

Comment:
Aren't the `kind` and `type` properties the same? Can we remove the `kind` property? I can't find it in the latest `development` branch.

Reply:
Typebox needs `kind` so we should refer `type` if anything

Reply:
> Typebox needs `kind`

I don't understand. The `kind` property is missing in the latest `development` branch of https://github.com/ubiquity-os/permit-generation. Why is it appeared suddenly?

Reply:
I have added it in this PR to create a Typebox union so that we are handling the two different permits type more efficiently than what we were doing before. Now the permit inputs are being validated statically and at runtime.

Can simplify it back to what we had before if need be

Reply:
Possible values for `type` are `ERC20` and `ERC721`.

What are the possible values for the `kind` property?

Reply:
Exactly the same, they are the same prop with a different name. 

I was disappointed I couldn't assign `type` as the discrim key and had to use `kind`. 

Effectively they are identical in terms of value and application.

Reply:
@gentlementlegen help. I still can't understand why we need 2 exactly same properties here.

Reply:
we don't @rndquu.

I'm removing `type`

Reply:
`kind` should not be needed technically, it is internal to typebox. I do not have the full context right now so I can't give you a proper answer but only one should be needed (and it is probably type).